File: TMessagesProj/src/main/java/uz/unnarsx/cherrygram/helpers/OnceBottomSheet.java
package uz.unnarsx.cherrygram.helpers;

import android.content.Context;

import org.telegram.ui.ActionBar.BottomSheet;

public class OnceBottomSheet extends BottomSheet {
    private static boolean shown = false;

    public OnceBottomSheet(Context context, boolean needFocus) {
        super(context, needFocus);
    }

    @Override
    public void show() {
        if (shown) {
            return;
        }
        shown = true;
        super.show();
    }
}

File: TMessagesProj/src/main/java/uz/unnarsx/cherrygram/prefviews/OnceBottomSheet.kt
package uz.unnarsx.cherrygram.prefviews

import android.content.Context
import org.telegram.ui.ActionBar.BottomSheet

open class OnceBottomSheet(context: Context?, needFocus: Boolean) :
    BottomSheet(context, needFocus) {
    override fun show() {
        if (shown) {
            return
        }
        shown = true
        super.show()
    }

    companion object {
        private var shown = false
    }
}